javascript/ql/src/semmle/javascript/security/dataflow/ExceptionXss.qll

```diff
@@ -24,16 +25,29 @@ module ExceptionXss {
     node = DataFlow::globalVarRef("console").getAMemberCall(_).getAnArgument()
   }
 
+  /**
+   * Holds if `t` is `null` or `undefined`.
+   */
+  private predicate isNullOrUndefined(InferredType t) {
+    t = TTNull() or
+    t = TTUndefined()
+  }
+
   /**
    * Holds if `node` can possibly cause an exception containing sensitive information to be thrown.
    */
   predicate canThrowSensitiveInformation(DataFlow::Node node) {
-    not isUnlikelyToThrowSensitiveInformation(node) and 
+    not isUnlikelyToThrowSensitiveInformation(node) and
     (
       // in the case of reflective calls the below ensures that both InvokeNodes have no known callee.
-      forex(DataFlow::InvokeNode call | node = call.getAnArgument() | not exists(call.getACallee()))
+      forex(DataFlow::InvokeNode call | call.getAnArgument() = node | not exists(call.getACallee()))
       or
       node.asExpr().getEnclosingStmt() instanceof ThrowStmt
+      or
+      exists(DataFlow::PropRef prop |
+        node = DataFlow::valueNode(prop.getPropertyNameExpr()) and
+        isNullOrUndefined(prop.getBase().analyze().getAType())
```

Comment:
Hm, I'm not sure about this condition. Our type inference is very conservative, and in particular is likely to infer `undefined` as a possible type of just about everything (apart from constants and sometimes SSA variables).

I think you may want to strengthen this to `forex(InferredType t | t = prop.getBase().analyze().getAType() | isNullOrUndefined(t))` to only cover those cases where we are quite sure the base is null-ish.

(Note that we don't want to use `forall`, since expressions in dead code sometimes don't have any types inferred for them.)